cogs/gpt_3_commands_and_converser.py

```diff
@@ -89,6 +89,17 @@ async def check_valid_roles(self, user, ctx):
             return False
         return True
 
+    @commands.Cog.listener()
+    async def on_member_join(self, member):
+        if self.welcome_message_enabled():
+            query = f"Please generate a welcome message for {member.name} who has just joined the server."
+            welcome_message = self.send_request(query, tokens=self.usage_service.count_tokens(query))
+            if not welcome_message:
+                welcome_message = EnvService.get_welcome_message()

```

Comment:
So if I'm understanding correctly, the intended behaviour is such that if GPT3 returns a valid welcome message, we use that, otherwise, we try to get it from .env, otherwise, we use our hardcoded default, correct?

Reply:
Correct. Try GPT, if that fails, check users "chosen" message in .env, and if that doesn't exist, use the hardcoded message.

Reply:
Got it!